src/MSBuild.Abstractions/MSBuildHelpers.cs

```diff
@@ -185,6 +185,26 @@ public static bool IsWinForms(IProjectRootElement projectRoot)
             return DesktopFacts.KnownWinFormsReferences.Any(reference => references.Contains(reference, StringComparer.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// 
+        ///Dtermines if a given project is a WinUI Project by looking at PackageReferences
+        /// </summary>
+        /// <param name="projectRoot"></param>
+        /// <returns></returns>
+        public static bool IsWinUI(IProjectRootElement projectRoot)
+        {
+            var packageReferences = projectRoot.ItemGroups.SelectMany(GetPackageReferences)?.Select(elem => elem.Include.Split(',').First());
+            var references = projectRoot.ItemGroups.SelectMany(GetReferences)?.Select(elem => elem.Include.Split(',').First());
+            var both = packageReferences.Union(references);
+            return WinUIFacts.KnownWinUIReferences.Any(reference => both.Contains(reference, StringComparer.OrdinalIgnoreCase));
+        }
+
+        /// <summary>
+        /// Gets all Reference items from a given item group.
+        /// </summary>
+        private static IEnumerable<ProjectItemElement> GetPackageReferences(ProjectItemGroupElement itemGroup) =>
```

Comment:
Its only used by MSBuild.Abstractions.MSBuildHelpers.IsWinUI, but it might be applicable in another instance. Or should I just combine it into the IsWinUI method?